internal/server/v1/warden/handler_test.go

```diff
@@ -0,0 +1,258 @@
+package warden
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	shieldv1beta1 "buf.build/gen/go/gotocompany/proton/protocolbuffers/go/gotocompany/shield/v1beta1"
+	"github.com/go-chi/chi/v5"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/goto/dex/internal/server/reqctx"
+	"github.com/goto/dex/internal/server/v1/warden/mocks"
+	shareMocks "github.com/goto/dex/mocks"
+)
+
+func TestHandler_teamList(t *testing.T) {
+	t.Run("should return 200 OK", func(t *testing.T) {
+		doer := mocks.NewDoer(t)
+		doer.EXPECT().Do(mock.Anything).Return(&http.Response{
+			StatusCode: http.StatusOK,
+			Body: io.NopCloser(bytes.NewBufferString(`{
+				"success": true,
+				"errors": [],
+				"data": {
+				"teams": [
+				{
+				"name": "data_fabric",
+				"created_at": "2023-08-25T03:52:13.548Z",
+				"updated_at": "2023-08-25T03:52:13.548Z",
+				"owner_id": 433,
+				"parent_team_identifier": "2079834a-05c4-420d-bfc8-44b934adea9f",
+				"identifier": "b5aea046-dab3-4dac-b1ea-e1eef423226b",
+				"product_group_name": "data_engineering",
+				"product_group_id": "2079834a-05c4-420d-bfc8-44b934adea9f",
+				"labels": null,
+				"short_code": "T394"
+				}
+				]
+				},
+				"status": "ok"
+				}`)),
+		}, nil)
+
+		r := chi.NewRouter()
+		r.Use(reqctx.WithRequestCtx())
+		r.Route("/dex/warden", Routes(nil, doer))
+
+		req, err := http.NewRequestWithContext(context.TODO(), http.MethodGet, "/dex/warden/users/me/warden_teams", nil)
+		assert.NoError(t, err)
+		req.Header.Add("X-Auth-Email", "test@email.com")
+
+		resp := httptest.NewRecorder()
+		r.ServeHTTP(resp, req)
+
+		assert.Equal(t, http.StatusOK, resp.Code)
+		assert.JSONEq(t, `{				"teams": [
+			{
+			"name": "data_fabric",
+			"created_at": "2023-08-25T03:52:13.548Z",
+			"updated_at": "2023-08-25T03:52:13.548Z",
+			"owner_id": 433,
+			"parent_team_identifier": "2079834a-05c4-420d-bfc8-44b934adea9f",
+			"identifier": "b5aea046-dab3-4dac-b1ea-e1eef423226b",
+			"product_group_name": "data_engineering",
+			"product_group_id": "2079834a-05c4-420d-bfc8-44b934adea9f",
+			"labels": null,
+			"short_code": "T394"
+			}
+			]}`, resp.Body.String())
+	})
+
+	t.Run("should return 404 when user email is not present in warden", func(t *testing.T) {
+		doer := mocks.NewDoer(t)
+		doer.EXPECT().Do(mock.Anything).Return(&http.Response{
+			StatusCode: http.StatusOK,
+			Body: io.NopCloser(bytes.NewBufferString(`{
+				"success": false,
```

Comment:
this is pretty hard to digest, let's care more about indentation and readability?

we can refactor it as well into variables if needed.

Reply:
simplified it using variable.